addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,292 @@
+                if po_line.order_id.state in ['draft', 'sent', 'to approve']:
+                    if id_per_supplier.get(po_line.partner_id.id):
+                        id_per_supplier[po_line.partner_id.id].append(po_line)
+                    else:
+                        id_per_supplier[po_line.partner_id.id] = [po_line]
+
+            #generate po based on supplier and cancel all previous RFQ
+            for supplier, product_line in id_per_supplier.items():
+                #copy a quotation for this supplier and change order_line then validate it
+                quotation_id = self.env['purchase.order'].search([('requisition_id', '=', tender.id), ('partner_id', '=', supplier)], limit=1)[0]
+                vals = self._prepare_po_from_tender()
+                new_po = quotation_id.copy(default=vals)
+                #duplicate po_line and change product_qty if needed and associate them to newly created PO
+                for line in product_line:
+                    vals = self._prepare_po_line_from_tender(line, new_po)
+                    vals['order_id'] = vals['order_id'].id
+                    line.copy(default=vals)
+                #use workflow to set new PO state to confirm
+                new_po.button_confirm()
+
+            #cancel other orders
+            self.cancel_unconfirmed_quotations()
+
+            #set tender to state done
+            # self.signal_workflow([tender.id], 'done')
+            tender.state = 'done'
+        return True
+
+    def cancel_unconfirmed_quotations(self):
+        #cancel other orders
+        for quotation in self:
+            if quotation.state in ['draft', 'sent', 'to approve']:
+                quotation.button_cancel()
+                quotation.message_post(body=_('Cancelled by the call for tenders associated to this request for quotation.'))
+        return True
+
+
+class PurchaseRequisitionLine(models.Model):
+    _name = "purchase.requisition.line"
+    _description = "Purchase Requisition Line"
+    _rec_name = 'product_id'
+
+    product_id = fields.Many2one('product.product', string='Product', domain=[('purchase_ok', '=', True)])
+    product_uom_id = fields.Many2one('product.uom', string='Product Unit of Measure')
+    product_qty = fields.Float(string='Quantity', digits_compute=dp.get_precision('Product Unit of Measure'))
+    requisition_id = fields.Many2one('purchase.requisition', string='Call for Tenders', ondelete='cascade')
+    company_id = fields.Many2one('res.company', related='requisition_id.company_id', string='Company', store=True, readonly=True, default=lambda self: self.env['res.users']._get_company())
+    account_analytic_id = fields.Many2one('account.analytic.account', string='Analytic Account',)
+    schedule_date = fields.Date(string='Scheduled Date')
+
+    @api.multi
+    def onchange_product_id(self, product_id, product_uom_id, parent_analytic_account, analytic_account, parent_date, date):
```

Comment:
Use @api.onchange instead of @api.multi and also refactor the method accordingly .